linera-chain/src/inbox.rs

```diff
@@ -243,7 +243,10 @@ where
     ViewError: From<<MemoryContext<()> as linera_views::common::Context>::Error>,
 {
     pub async fn new() -> Self {
-        let guard = Arc::new(Mutex::new(BTreeMap::new())).lock_arc().await;
+        let guard = Arc::new(Mutex::new(BTreeMap::new()))
+            .lock_arc()
+            .now_or_never()
+            .expect("a guard");
```

Comment:
Is this equivalent? Then we don't need the `futures` dependency.

```suggestion
        let guard = Arc::new(Mutex::new(BTreeMap::new()))
            .try_lock_arc()
            .expect("a guard");
```